### PR TITLE
TestData: attach labels to series results

### PR DESCRIPTION
**Pull request by ryantxu**
As we are messing with more data transformations, we need to have better tests that deal with the labels.  This lets us pass in prometheus style labels from the query editor:

image

and get the results as tags:
image

ok -- images are from different queries, but you get the idea :)

### CODE REVIEW

File: pkg/tsdb/testdata/scenarios.go [hunk 1]
```diff
@@ -517,9 +521,33 @@ func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResu
...
+	re.ReplaceAllFunc(byt, func(s []byte) []byte {
+		exp := string(s)
+		idx := strings.Index(exp, "=")
+		key := exp[:idx]
+		val := util.TrimQuotes(exp[idx+1:])
+		tags[key] = val
+		return s
+	})
+	for _, series := range queryRes.Series {
+		series.Tags = tags
```
Comment (torkelo) [inline pkg/tsdb/testdata/scenarios.go]: I wonder what would break if we change this to Labels :)   (in json response).

File: pkg/util/strings.go [hunk 2]
```diff
@@ -36,6 +36,16 @@ func SplitString(str string) []string {
 	return regexp.MustCompile("[, ]+").Split(str, -1)
 }
 
+// TrimQuotes returns a copy without the surrounding quote marks
+func TrimQuotes(s string) string {
+	if len(s) >= 2 {
```
Comment (kylebrandt) [inline pkg/util/strings.go]: strings.Trim , or  strings.TrimPrefix and strings.TrimSuffix depending on if you care about nested quotes: https://play.golang.org/p/r3Gr_f9G2xn

File: pkg/tsdb/testdata/scenarios.go [hunk 3]
```diff
@@ -517,9 +521,33 @@ func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResu
...
 	return queryRes
 }
 
+func attachLabels(query *tsdb.Query, queryRes *tsdb.QueryResult) {
+	labelText := query.Model.Get("labels").MustString("")
+	if labelText == "" {
+		return
+	}
+
+	re := regexp.MustCompile(`\b(\w+)(!?=~?)"([^"\n]*?)"`)
```
Comment (kylebrandt) [inline pkg/tsdb/testdata/scenarios.go]: Can concat the strings of the regex to make it more readable by adding comments, see https://stackoverflow.com/a/24589824/107156 . 

Also the regex should be a constant outside of the function (to only compile once).

**Comment by kylebrandt** [comment]
For user input, having the curly braces at start and end doesn't seem helpful? Except maybe in a copy / paste scenario.

File: pkg/tsdb/testdata/scenarios.go [hunk 4]
```diff
@@ -517,9 +521,33 @@ func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResu
...
+	tags := make(map[string]string)
+
+	byt := []byte(labelText)
+	re.ReplaceAllFunc(byt, func(s []byte) []byte {
+		exp := string(s)
+		idx := strings.Index(exp, "=")
```
Comment (kylebrandt) [inline pkg/tsdb/testdata/scenarios.go]: strings.SplitN is a little clearer IMO than slice indexing.

**Review by kylebrandt — CHANGES_REQUESTED** [review]
as per comments

**Comment by kylebrandt** [comment]
As a side note, we do have thing for inputting tags, e.g. https://grafana.com/docs/features/datasources/opentsdb/#query-editor , but I don't like it and would rather have the string :-) 

In some sense the prom tag model there is limiting in that the tag key can't have spaces. If we want to simulate SQL columns titles, they may have a space.

File: pkg/tsdb/testdata/scenarios.go (hunk 1 above)
Comment (ryantxu) [reply pkg/tsdb/testdata/scenarios.go]: it is also in gRPC, so I don't really want to think about it ;)

The backend signatures should stay the same for TableData and SeriesData...  when we add DataFrame, we can start to migrate them all to a consistent layout

File: pkg/util/strings.go (hunk 2 above)
Comment (ryantxu) [reply pkg/util/strings.go]: 👍

**Comment by ryantxu** [comment]
@kylebrandt -- thanks for the review.  I replaced the regex stuff (that I did not understand, just copy pasted!) with a more simple string splitting approach

the only downside to this is that tags can not contain `,` -- whatever

File: pkg/tsdb/testdata/scenarios.go (hunk 4 above)
Comment (ryantxu) [reply pkg/tsdb/testdata/scenarios.go]: got rid of regex and using SplitN

**Review by torkelo — APPROVED** [review]

**Review by kylebrandt — APPROVED** [review]

**Pull request merged**